tests/unit/preprocessor/RandomFourierGaussPreproc_unittest.cc

```diff
@@ -0,0 +1,66 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Author: Nanubala Gnana Sai
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/kernel/GaussianKernel.h>
+#include <shogun/mathematics/NormalDistribution.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/preprocessor/RandomFourierGaussPreproc.h>
+#include <math.h>
+using namespace shogun;
+
+class RFGPTest : public ::testing::Test
+{
+public:
+	virtual void SetUp()
+	{
+		SGMatrix<float64_t> mat(num_features, num_vectors);
+		linalg::range_fill(mat, 1.0);
+		auto features = std::make_shared<DenseFeatures<float64_t>>(mat);
+
+		preproc = std::make_shared<RandomFourierGaussPreproc>();
+		preproc->put(param0, seed);
+		preproc->put(param1, target_dim);
+		preproc->put(param2, log_width);
+		preproc->fit(features);
+	}
+	virtual void TearDown()
+	{
+	}
+
+protected:
+	const int32_t seed = 100;
+	const index_t num_vectors = 5;
+	const index_t num_features = 3;
+	const index_t target_dim = 300;
+	const float64_t log_width = log(100);
+	const float64_t epsilon = 0.1;
+	static constexpr std::string_view param0 = "seed";
+	static constexpr std::string_view param1 = "dim_output";
+	static constexpr std::string_view param2 = "log_width";
+	std::shared_ptr<RandomFourierGaussPreproc> preproc;
+};
+
+TEST_F(RFGPTest, apply_to_vectors)
+{
+	SGMatrix<float64_t> vectors(num_features, 2);
+	linalg::range_fill(vectors, 1.0);
+	auto feats = std::make_shared<DenseFeatures<float64_t>>(vectors);
+	auto preprocessed = std::static_pointer_cast<DenseFeatures<float64_t>>(
+	    preproc->transform(feats));
+
+	auto result_rff = linalg::dot(
+	    preprocessed->get_feature_vector(0),
```

Comment:
rather than this, just use `get` to extract the feature matrix, and then call `get_column` ... no need to use the DenseFeatures API

Reply:
mmmm i wouldn't use getter.... in c++ it's super brittle and ugly to see this in c++ code... and as soon as the parameter gets renamed the only way to get error is runtime.... while this should be popping up compile time.